discovery_engine_core/core/src/document.rs

```diff
@@ -93,13 +93,14 @@ pub struct Document {
     pub resource: NewsResource,
 }
 
-impl TryFrom<(Article, StackId, Embedding)> for Document {
+impl TryFrom<(GenericArticle, StackId, Embedding)> for Document {
```

Comment:
Yeah, I didn't want to continue this cascading change, as I feel this PR is big enough already and this is just a nice-to-have. Do. you feel strongly that this is 🔧 ? To me this seems like something we could definitely do, but more of a 🌱 .

Reply:
I would say 🔧  because it causes dead code and maybe even bugs because the code that handles the error case is unreachable.

Reply:
Mmh, I see the point. I think that a fair share of our error handling code is effectively dead code, because of errors that never actually occur in practice, or because we're not bothering to write tests for the error cases. I wasn't able to properly verify this, but the preliminary results of running `llvm-cov` do suggest that this is the case.

That being said, I see that this is not a great argument for creating more dead code, though I would not have put error handling that's not being hit in that category. Now I'm wondering, though, whether or not I need to re-think that 😅

Reply:
Ok then let's do it in a follow-up pr